CMakeLists.txt

```diff
@@ -5,6 +5,7 @@ if(COMMAND cmake_policy)
 endif(COMMAND cmake_policy)
 
 set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
```

Comment:
This should be tested for first:
https://cmake.org/cmake/help/latest/module/CheckIPOSupported.html#module:CheckIPOSupported

LTO in general increases build times, can expose bugs in our code and debuggers ... It's probably better to have this as an off-by-default option, so it's easy to enable for those who want it.